dashboard/app/models/ability.rb

```diff
@@ -184,6 +184,11 @@ def initialize(user)
     if user.admin?
       can :manage, :all
 
+      # We don't want admins to accidentally try to index _all_ existing
+      # sections so constrain their access.
+      cannot :index, Section
+      can :index, Section, user_id: user.id
```

Comment:
Also, read [Ability Precedence](https://github.com/ryanb/cancan/wiki/Ability-Precedence) in the CanCan docs to understand why the 'contradictions' in this specification work.